Meta/run.py

```
@@ -0,0 +1,902 @@
+
+
+def setup_boot_drive(config: Configuration):
+    provided_nvme_enable = environ.get("SERENITY_NVME_ENABLE")
+    if provided_nvme_enable is not None:
+        config.nvme_enable = provided_nvme_enable == "1"
+    provided_usb_boot_enable = environ.get("SERENITY_USE_SDCARD")
+    if provided_usb_boot_enable is not None:
+        config.sd_enable = provided_usb_boot_enable == "1"
+    provided_usb_boot_enable = environ.get("SERENITY_USE_USBDRIVE")
+    if provided_usb_boot_enable is not None:
+        config.usb_boot_enable = provided_usb_boot_enable == "1"
+
+    # NVMe does not work under MicroVM/ISA PC
+    if config.machine_type in [MachineType.MicroVM, MachineType.ISAPC]:
+        config.nvme_enable = False
+
+    if config.architecture == Arch.Aarch64:
+        config.boot_drive = f"file={config.disk_image},if=sd,format=raw,id=disk"
+    elif config.nvme_enable:
+        config.boot_drive = f"file={config.disk_image},format=raw,index=0,media=disk,if=none,id=disk"
+        config.add_devices(
+            [
+                "i82801b11-bridge,id=bridge4",
+                "sdhci-pci,bus=bridge4",
+                "nvme,serial=deadbeef,drive=disk,bus=bridge4,logical_block_size=4096,physical_block_size=4096",
+            ]
+        )
+        config.kernel_cmdline.append("root=nvme0:1:0")
+    elif config.sd_enable:
+        config.boot_drive = f"id=sd-boot-drive,if=none,format=raw,file={config.disk_image}"
+        config.add_devices(["sdhci-pci", "sd-card,drive=sd-boot-drive"])
+        config.kernel_cmdline.append("root=sd2:0:0")
+    elif config.usb_boot_enable:
+        config.boot_drive = f"if=none,id=usbstick,format=raw,file={config.disk_image}"
+        config.add_device("usb-storage,drive=usbstick")
+        # FIXME: Find a better way to address the usb drive
+        config.kernel_cmdline.append("root=block3:0")
+    else:
+        config.boot_drive = f"file={config.disk_image},format=raw,index=0,media=disk,id=disk"
```

My one gripe with this is that we may silently swallow provided argument
Can you make a BootDrive enum instead?
à la
```
SERENITY_BOOT_DRIVE=usb Meta/serenity.py run
```
Not sure if we need to keep the old `SERENITY_USE_(USBDRIVE|SDCARD)` then
Also as said above, warn if the machine type does not support the selected boot medium

---

The default boot medium would be decided by architecture and machine type

---

We can do that later. I will leave a FIXME.